src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -658,6 +720,45 @@ def schema_query(self) -> str:
     def children_plan_nodes(self) -> List[Union["Selectable", SnowflakePlan]]:
         return [self.from_]
 
+    @property
```

Comment:
Should we override it for SelectSnowflakePlan by reading complexity score from snowflake_plan directly?

Reply:
yep. added